extensions/sql-database-projects/src/controllers/projectController.ts

```diff
@@ -649,134 +607,64 @@ export class ProjectsController {
-				{
-					canPickMany: false,
-					placeHolder: constants.extractDatabaseSelection
-				}))?.label;
+			let newProjFolderUri = model.filePath;
 
-			if (!database) {
-				throw new Error(constants.databaseSelectionRequired);
-			}
-		}
+			const newProjFilePath = await this.createNewProject(model.projName, vscode.Uri.file(newProjFolderUri), true);
+			model.filePath = path.dirname(newProjFilePath);
+			this.setFilePath(model);
 
-		model.database = database;
-		model.serverId = connectionId;
+			const project = await Project.openProject(newProjFilePath);
+			await this.createProjectFromDatabaseApiCall(model); // Call ExtractAPI in DacFx Service
+			let fileFolderList: string[] = model.extractTarget === mssql.ExtractTarget.file ? [model.filePath] : await this.generateList(model.filePath); // Create a list of all the files and directories to be added to project
 
-		return model;
-	}
+			await project.addToProject(fileFolderList); // Add generated file structure to the project
 
-	private getConnectionProfileFromContext(context: azdata.IConnectionProfile | any): azdata.IConnectionProfile | undefined {
-		if (!context) {
-			return undefined;
+			// add project to workspace
+			workspaceApi.showProjectsView();
+			await workspaceApi.addProjectsToWorkspace([vscode.Uri.file(newProjFilePath)]);
 		}
-
-		// depending on where import new project is launched from, the connection profile could be passed as just
-		// the profile or it could be wrapped in another object
-		return (<any>context).connectionProfile ? (<any>context).connectionProfile : context;
-	}
-
-	private async getProjectName(dbName: string): Promise<string> {
-		let projName = await vscode.window.showInputBox({
-			prompt: constants.newDatabaseProjectName,
-			value: newProjectTool.defaultProjectNameFromDb(dbName)
-		});
-
-		projName = projName?.trim();
-
-		if (!projName) {
-			throw new Error(constants.projectNameRequired);
+		catch (err) {
+			vscode.window.showErrorMessage(utils.getErrorMessage(err));
 		}
-
-		return projName;
 	}
 
-	private mapExtractTargetEnum(inputTarget: any): mssql.ExtractTarget {
-		if (inputTarget) {
-			switch (inputTarget) {
-				case constants.file: return mssql.ExtractTarget['file'];
-				case constants.flat: return mssql.ExtractTarget['flat'];
-				case constants.objectType: return mssql.ExtractTarget['objectType'];
-				case constants.schema: return mssql.ExtractTarget['schema'];
-				case constants.schemaObjectType: return mssql.ExtractTarget['schemaObjectType'];
-				default: throw new Error(constants.invalidInput(inputTarget));
-			}
-		} else {
-			throw new Error(constants.extractTargetRequired);
+	public setFilePath(model: ImportDataModel) {
+		if (model.extractTarget === mssql.ExtractTarget.file) {
+			model.filePath = path.join(model.filePath, model.projName + '.sql'); // File extractTarget specifies the exact file rather than the containing folder
```

Comment:
is model.filePath the path to a file or directory?

Reply:
It is based on the extract target. If it is "File", then model.filePath is a sql file otherwise (in all other scenarios) it is a directory.